lib/JSON/Validator.pm

```diff
@@ -196,141 +165,114 @@ sub _load_schema_from_text {
 }
 
 sub _load_schema_from_url {
-  my ($self, $url, $namespace) = @_;
+  my ($self, $url) = @_;
   my $cache_path = $self->cache_paths->[0];
-  my $cache_file = Mojo::Util::md5_sum($namespace);
-  my $tx;
+  my $cache_file = Mojo::Util::md5_sum("$url");
+  my ($err, $tx);
 
   for (@{$self->cache_paths}) {
     my $path = path $_, $cache_file;
     next unless -r $path;
     warn "[JSON::Validator] Loading cached file $path\n" if DEBUG;
-    return $path->slurp;
+    return $self->_load_schema_from_text(\$path->slurp);
   }
 
   $tx = $self->ua->get($url);
-  die $tx->error->{message} if $tx->error;
+  $err = $tx->error && $tx->error->{message};
+  Carp::confess("GET $url == $err") if DEBUG and $err;
+  die "[JSON::Validator] GET $url == $err" if $err;
 
   if ($cache_path and $cache_path ne $BUNDLED_CACHE_DIR and -w $cache_path) {
     $cache_file = path $cache_path, $cache_file;
-    warn "[JSON::Validator] Caching $namespace to $cache_file\n" unless $ENV{HARNESS_ACTIVE};
+    warn "[JSON::Validator] Caching $url to $cache_file\n" unless $ENV{HARNESS_ACTIVE};
     $cache_file->spurt($tx->res->body);
   }
 
-  return $tx->res->body;
+  return $self->_load_schema_from_text(\$tx->res->body);
 }
 
-sub _default_id {
-  state $id = 0;
-  return sprintf 'http://local/json-validator/default-id-%s', ++$id;
-}
+sub _reset { delete $_[0]->{refs}; $_[0] }
```

Comment:
Not sure if it's needed to delete the whole "refs" key or if it's enough to just delete `$_[0]->{refs}{""}`.